components/listingsPage/__tests__/Filters.test.js

```diff
@@ -0,0 +1,69 @@
+import { shallow } from "enzyme";
+import Filters  from "../filters";
+import { FormDiv, ToggleIcon } from "../utils";
+import { findByTestAtrr } from "../../../Utils/test";
+
+const mockDispatch = jest.fn();
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+  useDispatch: () => mockDispatch
+}));
+
+jest.mock("react-i18next", () => ({
+    withTranslation: () => (Component) => {
+      Component.defaultProps = { ...Component.defaultProps, t: () => [] };
+      return Component;
+    },
+  }));
+
+
+
+describe("Filters Component Tests", () => {
+
+    let filters;
+    const ToggleMobile = jest.fn();
```

Comment:
I think the `ToggleMobile` method doesn’t need to be mocked here because it doesn’t get directly used in the test. We could though test that it is called when `ToggleIcon` is clicked.

Reply:
Cant just check if it is clicked because it is a functional component and enzyme is being used to render the component. The only did i noticed there is that the mock is being done too early and the passed even to test that does not require it. So, I have separated the toggle click into its own suite. to avoid unnecessary function being passed around. Thanks.

Reply:
OK, makes sense. 👍